python/packages/autogen-core/src/autogen_core/models/_model_client.py

```diff
@@ -222,6 +223,7 @@ async def create(
         Args:
             messages (Sequence[LLMMessage]): The messages to send to the model.
             tools (Sequence[Tool | ToolSchema], optional): The tools to use with the model. Defaults to [].
+            tool_choice (Optional[Sequence[Union[str, Tool]]], optional): A list of tool names or Tool objects to restrict the model's choice to. Defaults to None.
```

Comment:
By default, the tool choice should be auto as `Literal["auto"]` type, which means model can choose any one supplied in the `tools` parameter.

Let's use the type hint `Tool | Literal['auto'] | None = "auto"` for simplicity. Remove the option to use the string name of the tool in order to avoid confusion with "auto". And make sure only one tool can be used as a choice as model API providers don't support multiple tool choices.  

Apply the same changes to `create_stream`.